src/Evolution/DiscontinuousGalerkin/InboxTags.hpp

```diff
@@ -0,0 +1,126 @@
+#include "Parallel/InboxInserters.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace evolution::dg::Tags {
+/*!
+ * \brief The inbox tag for boundary correction communication and DG-subcell
+ * ghost zone cells.
+ *
+ * The first element in the tuple is the mesh of the neighboring element's
+ * face (not the mortar mesh!), the second is the variables at the ghost zone
+ * cells for finite difference/volume reconstruction, and the third is the data
+ * on the mortar needed for computing the boundary corrections (e.g. fluxes,
+ * characteristic speeds, conserved variables). he fourth element of the tuple
+ * is the TimeStepId beyond which the boundary terms are no longer valid, when
+ * using local time stepping.
+ *
+ * The TimeStepId is the neighboring element's next time step. When using local
+ * time stepping, the neighbor's boundary data is valid up until this time,
+ * which may include multiple local time steps. By receiving and storing the
+ * neighbor time step, the local element knows whether or not it should remove
+ * boundary data and expect new data to be sent from the neighbor.
+ *
+ * The ghost cell data (second element of the tuple) will be valid whenever
+ * DG-subcell scheme is being used. Whenever a DG-subcell scheme is being used,
+ * elements using DG and not FD/FV always send both the ghost cells and boundary
+ * correction data together. Elements using FD/FV send the ghost cells first
+ * followed by the boundary correction data once the element has received all
+ * neighbor ghost cell data. Note that the second send/receive only modifies the
+ * flux and the TimeStepId used for the flux validity range.
+ *
+ * When only a DG scheme (not a DG-subcell scheme) is used the ghost cell data
+ * will never be valid.
+ *
+ * In the DG-subcell scheme this tag is used both for communicating the ghost
+ * cell data needed for the FD/FV reconstruction step and the data needed for
+ * the boundary corrections. An element using a DG scheme will send both the
+ * ghost cells and the boundary correction terms simultaneously, while an
+ * element using FD/FV will first send the ghost cell data. The element
+ * receiving the ghost cells will store the data in this inbox tag and be able
+ * to do reconstruction once all neighboring ghost cells are received. In the
+ * mean time, as boundary correction data is received from neighbors it is added
+ * to this inbox tag. The inbox tag is not erased until after the boundary
+ * correction computation is complete. The reason for this design choice is that
+ * it is mostly the number of communications that dominates cost, not the size
+ * of the communication.
+ */
+template <size_t Dim>
+struct BoundaryCorrectionAndGhostCellsInbox {
+  using stored_type =
+      std::tuple<Mesh<Dim - 1>, std::optional<std::vector<double>>,
+                 std::optional<std::vector<double>>, ::TimeStepId>;
+
+ public:
+  using temporal_id = TimeStepId;
+  using type = std::map<
```

Comment:
All the inbox tags use a `map` for the time. This is a decision made long before we even went to GitHub, i think. Since the time/temporal id are ordered it seems reasonable to use a `map`, but we also don't rely on that ordering so it could be changed. However, changing it should be done as a more global change. Does that make sense?

Reply:
yes, just wanted to make sure I wasn't missing anything important in which case I'd have asked for a comment

Reply:
The original reason for `std::map` was that some of the LTS code iterates over the map in its inbox in order.  Probably that just propagated everywhere else.

Reply:
ah right, that makes sense. I see down side to using it everywhere :shrug: